specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/environmentSpecification.json

```diff
@@ -0,0 +1,414 @@
+        ],
+        "operationId": "EnvironmentSpecificationVersions_Get",
+        "x-ms-examples": {
+          "Get Environment Specification Version.": {
+            "$ref": "./examples/getEnvironmentSpecificationVersion.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "machineLearningServices.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/APIVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/EnvironmentNameParameter"
+          },
+          {
+            "$ref": "#/parameters/EnvironmentVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          },
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/EnvironmentSpecificationVersionResource"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "EnvironmentSpecification"
+        ],
+        "operationId": "EnvironmentSpecificationVersions_Delete",
+        "x-ms-examples": {
+          "Get Environment Specification Version.": {
+            "$ref": "./examples/deleteEnvironmentSpecificationVersion.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "machineLearningServices.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/APIVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/EnvironmentNameParameter"
+          },
+          {
+            "$ref": "#/parameters/EnvironmentVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          },
+          "200": {
+            "description": "Success"
+          },
+          "204": {
+            "description": "Success"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/environments/{environmentName}/versions/latest": {
```

Comment:
This URI implies "latest" is the name of one of the versions which is not accurate.  Please instead use a query parameter on the GET /versions API to indicate that only the latest should be returned.

Reply:
Follow up question: What about Azure Billing ARM resource? In https://docs.microsoft.com/en-us/rest/api/billing/2017-04-24-preview/invoices/getlatest the route is used to get the latest version instead of a query parameter. Is this not ARM best practice?

Reply:
at-ing @pilor, since I'm not sure how notifications on un-resolved comments works... sorry if this is a double ping